src/compiler/checker.ts

```diff
@@ -3995,13 +3995,23 @@ namespace ts {
             if (!(symbol && !isPropertyOrMethodDeclarationSymbol(symbol))) {
                 return undefined;
             }
+            const links = getSymbolLinks(symbol);
+            const cache = (links.accessibleChainCache ||= new Map());
+            // Go from enclosingDeclaration to the first scope we check, so the cache is keyed off the scope and thus shared more
+            const firstRelevantLocation = forEachSymbolTableInScope(enclosingDeclaration, (_, __, node) => node);
+            const key = `${useOnlyExternalAliasing ? 0 : 1}|${firstRelevantLocation && getNodeId(firstRelevantLocation)}|${meaning}`;
+            if (cache.has(key)) {

```

Comment:
Even better is to set the result as cache.set(result || false) and then get becomes:
```ts
const existing = cache.get(key);
if (existing !== undefined) return existing || undefined;
```

Reply:
Neither of these are particularly idiomatic (plus the first adds extra allocations and indirections while the second adds polymorphic accesses to avoid those), nor are they as likely to be optimized by the runtime in the future (a map `get` immediately following a checked `has` is enormously common). Moreover, even in this example where I know the cache is often hit, the difference doesn't result in any measurable performance improvement (I tried, and got the same overall times out). Given all that, I'd prefer not to change it.